app/src/main/java/com/android/streetworkapp/ui/train/TrainHubScreen.kt

```diff
@@ -0,0 +1,214 @@
+                    "Coach" -> R.drawable.training_coach
+                    "Challenge" -> R.drawable.training_challenge
+                    else -> R.drawable.pushup
+                  }
+                },
+                isSelected = { it == selectedType },
+                testTagPrefix = "Role_")
+
+            Spacer(modifier = Modifier.height(16.dp))
+            HorizontalDivider(
+                modifier = Modifier.padding(vertical = 16.dp).testTag("Divider"),
+                thickness = 1.dp,
+                color = BORDER_COLOR)
+
+            // Activity selection
+            SectionTitle(title = "Choose your activity", testTag = "ActivitySelectionTitle")
+            SelectionGrid(
+                items =
+                    listOf(
+                            "Push-ups" to false,
+                            "Dips" to false,
+                            "Burpee" to false,
+                            "Lunge" to false,
+                            "Planks" to true,
+                            "Handstand" to true,
+                            "Front lever" to true,
+                            "Flag" to true,
+                            "Muscle-up" to false)
+                        .map { it.first },
+                buttonSize = buttonSize,
+                onItemSelected = { activity ->
+                  selectedActivity =
+                      listOf(
+                              "Push-ups" to false,
+                              "Dips" to false,
+                              "Burpee" to false,
+                              "Lunge" to false,
+                              "Planks" to true,
+                              "Handstand" to true,
+                              "Front lever" to true,
+                              "Flag" to true,
+                              "Muscle-up" to false)
+                          .first { it.first == activity }
+                },
+                getItemImageRes = { R.drawable.pushup },
+                isSelected = { selectedActivity?.first == it },
+                testTagPrefix = "Activity_")
+          }
+
+          // Confirm button
+          Box(modifier = Modifier.fillMaxWidth()) {
+            ConfirmButton(
+                modifier = Modifier.align(Alignment.Center),
+                onClick = {
+                  selectedType?.let { type ->
+                    selectedActivity?.let { (activity, isTimeDependent) ->
+                      when (type) {
+                        "Solo" -> navigationActions.navigateToSoloScreen(activity, isTimeDependent)
+                        "Coach" ->
+                            navigationActions.navigateToCoachScreen(activity, isTimeDependent)
+                        "Challenge" ->
+                            navigationActions.navigateToChallengeScreen(activity, isTimeDependent)
+                      }
+                    }
+                  }
+                })
```

Comment:
You might want to abstract away some of that logic (Confirm Button)to clean the ui from logical components. This can also help you to test things in the future.

Reply:
Done